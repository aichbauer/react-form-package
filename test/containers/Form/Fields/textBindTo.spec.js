import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import {
  Form,
  Field,
  Button,
} from '../../../../src';
import { textInputAllRules } from '../../../fixtures';

configure({ adapter: new Adapter() });

test('Render Form with text input | text input all rules', () => {
  const mock = jest.fn();
  const event = {
    target: {
      id: 'housenumbers',
      value: '1-2',
    },
  };
  const rules = {
    type: 'text',
    min: '-1',
    max: '-1',
    required: false,
  };

  const myComponent = (
    <Form>
      <Field
        id="housenumbers"
        type="text"
        bindTo="households"
        placeholder="2-7"
        bindToCallback={mock}
      />
      <Field
        id="households"
        type="number"
        placeholder="6"
      />
    </Form>
  );

  const tree = renderer.create(myComponent).toJSON();
  const wrapper = mount(myComponent);

  wrapper.find('input#housenumbers').simulate('focus');
  wrapper.find('input#housenumbers').simulate('change', event, rules);
  wrapper.find('input#housenumbers').simulate('blur');

  expect(mock).toBeCalledTimes(1);
  expect(tree).toMatchSnapshot();
});
