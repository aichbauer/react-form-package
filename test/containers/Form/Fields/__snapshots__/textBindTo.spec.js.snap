// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Form with text input | text input all rules 1`] = `
<form>
  <input
    bindTo="households"
    bindToCallback={
      [MockFunction] {
        "calls": Array [
          Array [
            "1-2",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    className="rfp-input"
    id="housenumbers"
    invalid={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="2-7"
    type="text"
    valid={false}
    value=""
  />
  <input
    className="rfp-input"
    id="households"
    invalid={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="6"
    type="number"
    valid={false}
    value=""
  />
</form>
`;
